lib/Entities/SubscriberEntity.php

```diff
@@ -0,0 +1,350 @@
+<?php
+
+namespace MailPoet\Entities;
+
+use DateTimeInterface;
+use MailPoet\Doctrine\EntityTraits\AutoincrementedIdTrait;
+use MailPoet\Doctrine\EntityTraits\CreatedAtTrait;
+use MailPoet\Doctrine\EntityTraits\DeletedAtTrait;
+use MailPoet\Doctrine\EntityTraits\UpdatedAtTrait;
+use MailPoetVendor\Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Entity()
+ * @ORM\Table(name="subscribers")
+ */
+class SubscriberEntity {
+  // statuses
+  const STATUS_BOUNCED = 'bounced';
+  const STATUS_INACTIVE = 'inactive';
+  const STATUS_SUBSCRIBED = 'subscribed';
+  const STATUS_UNCONFIRMED = 'unconfirmed';
+  const STATUS_UNSUBSCRIBED = 'unsubscribed';
+
+  use AutoincrementedIdTrait;
+  use CreatedAtTrait;
+  use UpdatedAtTrait;
+  use DeletedAtTrait;
+
+  /**
+   * @ORM\Column(type="bigint")
+   * @var int|null
+   */
+  private $wp_user_id;
+
+  /**
+   * @ORM\Column(type="boolean")
+   * @var bool|null
+   */
+  private $is_woocommerce_user;
+
+  /**
+   * @ORM\Column(type="string")
+   * @var string
+   */
+  private $first_name;
+
+  /**
+   * @ORM\Column(type="string")
+   * @var string
+   */
+  private $last_name;
+
+  /**
+   * @ORM\Column(type="string")
+   * @var string
+   */
+  private $email;
+
+  /**
+   * @ORM\Column(type="string")
+   * @var string
+   */
+  private $status = self::STATUS_UNCONFIRMED;
+
+  /**
+   * @ORM\Column(type="string")
+   * @var string
+   */
+  private $subscribed_ip;
+
+  /**
+   * @ORM\Column(type="string")
+   * @var string|null
+   */
+  private $confirmed_ip;
+
+  /**
+   * @ORM\Column(type="datetimetz")
+   * @var DateTimeInterface|null
+   */
+  private $confirmed_at;
+
+  /**
+   * @ORM\Column(type="datetimetz")
+   * @var DateTimeInterface|null
+   */
+  private $last_subscribed_at;
+
+  /**
+   * @ORM\Column(type="text")
+   * @var string|null
+   */
+  private $unconfirmed_data;
+
+  /**
+   * @ORM\Column(type="string")
+   * @var string|null
+   */
+  private $source = 'unknown';
+
+  /**
+   * @ORM\Column(type="int")
+   * @var int|null
+   */
+  private $count_confirmations;
```

Comment:
This has default value 0 in DB so we should set it as default to property and remove `null` from the annotation. When changing annotation update also getter and setter.